src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -1646,40 +1648,95 @@ TEST_CASE("GradientConcatenateLayerTest", "[ANNLayerTest]")
  */
 TEST_CASE("SimpleLookupLayerTest", "[ANNLayerTest]")
 {
-  arma::mat output, input, delta, gradient;
-  Lookup<> module(10, 5);
+  const size_t vocabSize = 10;
+  const size_t embeddingSize = 2;
+  const size_t seqLength = 3;
+  const size_t batchSize = 4;
+
+  arma::mat output, input, gy, g, gradient;
+
+  Lookup<> module(vocabSize, embeddingSize);
   module.Parameters().randu();
 
   // Test the Forward function.
-  input = arma::zeros(2, 1);
-  input(0) = 1;
-  input(1) = 3;
+  input = arma::zeros(seqLength, batchSize);
+  for (size_t i = 0; i < input.n_elem; ++i)
+  {
+    int token = math::RandInt(1, vocabSize);
+    input(i) = token;
+  }
 
   module.Forward(input, output);
+  for (size_t i = 0; i < batchSize; ++i)
+  {
+    // The Lookup module uses index - 1 for the cols.
+    const double outputSum = arma::accu(module.Parameters().rows(
+        arma::conv_to<arma::uvec>::from(input.col(i)) - 1));
 
-  // The Lookup module uses index - 1 for the cols.
-  const double outputSum = arma::accu(module.Parameters().col(0)) +
-      arma::accu(module.Parameters().col(2));
+    REQUIRE(std::abs(outputSum - arma::accu(output.col(i))) <= 1e-5);
```

Comment:
In plain C the `abs()` function returns `int`, but in C++ there's an overload for `double`. So, this code is correct. Nevertheless, I prefer to use `fabs()` instead of `abs()`.
```suggestion
    REQUIRE(std::fabs(outputSum - arma::accu(output.col(i))) <= 1e-5);
```